qiskit/circuit/library/standard_gates/x.py

```diff
@@ -606,65 +554,13 @@ def __init__(
     _singleton_lookup_key = stdlib_singleton_key(num_ctrl_qubits=3)
 
     def _define(self):
-        """
-        gate c3sqrtx a,b,c,d
-        {
-            h d; cu1(pi/8) a,d; h d;
-            cx a,b;
-            h d; cu1(-pi/8) b,d; h d;
-            cx a,b;
-            h d; cu1(pi/8) b,d; h d;
-            cx b,c;
-            h d; cu1(-pi/8) c,d; h d;
-            cx a,c;
-            h d; cu1(pi/8) c,d; h d;
-            cx b,c;
-            h d; cu1(-pi/8) c,d; h d;
-            cx a,c;
-            h d; cu1(pi/8) c,d; h d;
-        }
-        """
+        """Default definition"""
         # pylint: disable=cyclic-import
-        from qiskit.circuit import QuantumCircuit, QuantumRegister
-        from .u1 import CU1Gate
-        from .h import HGate
-
-        angle = numpy.pi / 8
-        q = QuantumRegister(4, name="q")
-        rules = [
-            (HGate(), [q[3]], []),
-            (CU1Gate(angle), [q[0], q[3]], []),
-            (HGate(), [q[3]], []),
-            (CXGate(), [q[0], q[1]], []),
-            (HGate(), [q[3]], []),
-            (CU1Gate(-angle), [q[1], q[3]], []),
-            (HGate(), [q[3]], []),
-            (CXGate(), [q[0], q[1]], []),
-            (HGate(), [q[3]], []),
-            (CU1Gate(angle), [q[1], q[3]], []),
-            (HGate(), [q[3]], []),
-            (CXGate(), [q[1], q[2]], []),
-            (HGate(), [q[3]], []),
-            (CU1Gate(-angle), [q[2], q[3]], []),
-            (HGate(), [q[3]], []),
-            (CXGate(), [q[0], q[2]], []),
-            (HGate(), [q[3]], []),
-            (CU1Gate(angle), [q[2], q[3]], []),
-            (HGate(), [q[3]], []),
-            (CXGate(), [q[1], q[2]], []),
-            (HGate(), [q[3]], []),
-            (CU1Gate(-angle), [q[2], q[3]], []),
-            (HGate(), [q[3]], []),
-            (CXGate(), [q[0], q[2]], []),
-            (HGate(), [q[3]], []),
-            (CU1Gate(angle), [q[2], q[3]], []),
-            (HGate(), [q[3]], []),
-        ]
-        qc = QuantumCircuit(q)
-        for instr, qargs, cargs in rules:
-            qc._append(instr, qargs, cargs)
+        from qiskit.circuit import QuantumCircuit
 
-        self.definition = qc
+        self.definition = QuantumCircuit._from_circuit_data(
+            StandardGate.C3SX._get_definition(self.params), add_regs=True, name=self.name
+        )
```

Comment:
maybe add an image of the circuit?

Reply:
The definitions for the 3-controlled circuits are quite large and don't git on a single page. This is why I have provided the definitions for almost all circuits but not the very large ones.